test/test_sync_pending_bootstraps.cpp

```
@@ -127,25 +134,23 @@ TEST(Sync_PendingBootstrapStoreClear)
     changesets.emplace_back(2, 7, BinaryData(changeset_data.back()), 2, 1);
     changesets.back().original_changeset_size = 1024;
 
-    bool created_new_batch = false;
-    store.add_batch(2, progress, changesets, &created_new_batch);
-    CHECK(created_new_batch);
+    int64_t query_version = sub_store->get_latest().make_mutable_copy().commit().version();
+    store.add_batch(query_version, progress, changesets);
     CHECK(store.has_pending());
+    CHECK_EQUAL(SubscriptionSet::State::Bootstrapping, sub_store->get_latest().state());
 
-    auto pending_batch = store.peek_pending(1025);
+    auto pending_batch = store.peek_pending(*db->start_read(), 1025);
     CHECK_EQUAL(pending_batch.remaining_changesets, 0);
-    CHECK_EQUAL(pending_batch.query_version, 2);
+    CHECK_EQUAL(pending_batch.query_version, query_version);
     CHECK(pending_batch.progress);
     CHECK_EQUAL(pending_batch.changesets.size(), 2);
 
-    store.clear();
+    auto tr = db->start_write();
+    store.clear(*tr, query_version);
+    tr->commit();
 
+    CHECK_EQUAL(SubscriptionSet::State::Pending, sub_store->get_latest().state());
     CHECK_NOT(store.has_pending());
-    pending_batch = store.peek_pending(1024);
```

This is now an assertion failure rather than returning an empty result.